contracts/colony/ColonyDataTypes.sol

```diff
@@ -318,10 +318,12 @@ interface ColonyDataTypes {
   struct Domain {
     uint256 skillId;
     uint256 fundingPotId;
+    bool deprecated;
```

Comment:
Having trouble writing a test to demonstrate it right now (IMWSS doesn't play ball with the 'historical' deployment scripts right now, it seems), but while this change is fine standalone, in the context of an already deployed system, breaks anything that calls `getDomain` on a colony. That's a lot of extensions, but also the `ReputationMiningCycle`, so this can't be merged until that's solved (and, ideally, a test written).

Reply:
After some experimenting, I think going this way is okay, and it's only removing elements from such a struct that would cause a revert to occur? I used this to tinker:

```
pragma solidity >=0.7.0 <0.9.0;

contract Test1 {
  struct Domain {
    uint256 skillId;
    uint256 fundingPotId;
  }

  function x() public pure returns (Domain memory domain) {
    return Domain(1, 1);
  }
}

contract Test2 {
  struct Domain {
    uint256 skillId;
    uint256 fundingPotId;
    bool deprecated;
  }

    function x() public pure returns (Domain memory domain) {
        return Domain(2, 2, false);
    }
}

contract Test3 {
    function test(address a) public pure returns (uint256) {
        Test1.Domain memory d = Test1(a).x();
        return d.skillId;
    }
}

contract Test4 {
    function test(address a) public pure returns (uint256) {
        Test2.Domain memory d = Test2(a).x();
        return d.skillId;
    }
}
```

I would still very much like a test using our contracts demonstrating that's the case, though - bricking the mining cycle, in particular, is unacceptable.